src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine_MapCreation.cs

```diff
@@ -166,8 +167,19 @@ private async Task<ProjectMap> CreateProjectMapAsync(ConcurrentSet<ISymbol> symb
             }
         }
 
+        private async Task<SymbolGroup> DetermineSymbolGroupAsync(ISymbol searchSymbol)
+        {
+            if (!_options.Cascade)
+                return new SymbolGroup(searchSymbol, ImmutableArray.Create(searchSymbol));
+
+            var linkedSymbols = await SymbolFinder.FindLinkedSymbolsAsync(
+                searchSymbol, _solution, _cancellationToken).ConfigureAwait(false);
+
+            return new SymbolGroup(searchSymbol, linkedSymbols);
+        }
```

Comment:
this is the core change in the find-refs low level code.  Instead of having a LinkedSymbolFinder, we just have the core engine do the linking.  We then represent this as a first class concept (the 'symbol group').